buildSrc/src/main/kotlin/otel.javaagent-testing.gradle.kts

```diff
@@ -0,0 +1,111 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import io.opentelemetry.instrumentation.gradle.bytebuddy.ByteBuddyPluginConfigurator
+
+plugins {
+  id("net.bytebuddy.byte-buddy")
+
+  id("otel.instrumentation-conventions")
+  id("otel.shadow-conventions")
+}
+
+val toolingRuntime by configurations.creating {
+  isCanBeConsumed = false
+  isCanBeResolved = true
+}
+
+dependencies {
+  // Integration tests may need to define custom instrumentation modules so we include the standard
+  // instrumentation infrastructure for testing too.
+  compileOnly(project(":instrumentation-api"))
+  compileOnly(project(":javaagent-api"))
+  compileOnly(project(":javaagent-bootstrap"))
+  // Apply common dependencies for instrumentation.
+  compileOnly(project(":javaagent-extension-api")) {
+    // OpenTelemetry SDK is not needed for compilation
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk")
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk-metrics")
+  }
+  compileOnly(project(":javaagent-tooling")) {
+    // OpenTelemetry SDK is not needed for compilation
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk")
+    exclude(group = "io.opentelemetry", module = "opentelemetry-sdk-metrics")
+  }
+  compileOnly("net.bytebuddy:byte-buddy")
+  annotationProcessor("com.google.auto.service:auto-service")
+  compileOnly("com.google.auto.service:auto-service")
+  compileOnly("org.slf4j:slf4j-api")
+
+  testImplementation(project(":testing-common"))
+
+  testImplementation("org.testcontainers:testcontainers")
+
+  toolingRuntime(project(path = ":javaagent-tooling", configuration = "instrumentationMuzzle"))
```

Comment:
I'm going to try to rework the bytebuddy / muzzle plugins to move this to the other plugin since they shouldn't be needed for tests (tests just need to muzzle the test source sets for generating helper class names)